src/main/java/nextstep/sessions/service/SessionService.java

```diff
@@ -1,22 +1,54 @@
 package nextstep.sessions.service;
 
+import java.util.List;
+
 import nextstep.payments.domain.Payment;
 import nextstep.sessions.domain.data.Session;
+import nextstep.sessions.domain.data.type.ApprovalType;
+import nextstep.sessions.domain.data.type.SelectionType;
 import nextstep.sessions.domain.data.vo.Registration;
+import nextstep.sessions.domain.exception.SessionsException;
+import nextstep.sessions.repository.RegistrationRepository;
 import nextstep.sessions.repository.SessionRepository;
 import nextstep.users.domain.NsUser;
 
 public class SessionService {
 
     private final SessionRepository sessionRepository;
+    private final RegistrationRepository registrationRepository;
 
-    public SessionService(SessionRepository sessionRepository) {
+    public SessionService(SessionRepository sessionRepository, RegistrationRepository registrationRepository) {
         this.sessionRepository = sessionRepository;
+        this.registrationRepository = registrationRepository;
     }
 
     public void enroll(int sessionId, NsUser loginUser, Payment payment) {
-        Session session = sessionRepository.findSessionBySessionId(sessionId);
-        Registration registration = session.registration(loginUser, payment);
-        sessionRepository.saveRegistration(sessionId, registration);
+        Session session = sessionRepository.findSessionBySessionId(sessionId)
+            .orElseThrow(() -> new SessionsException("강의 정보가 없습니다."));
+        List<Registration> registrations = registrationRepository.findAllRegistrations(sessionId);
+        Session sessionWithRegistrations = session.with(registrations);
+        Registration registration = sessionWithRegistrations.registration(loginUser, payment);
+        registrationRepository.saveRegistration(sessionId, registration);
+    }
+
+    public void select(int registrationId) {
+        registration(registrationId).validateSelection();
+        registrationRepository.updateSelectionType(registrationId, SelectionType.SELECTION);
+    }
```

Comment:
객체에게 메시지를 보내서 값을 변경한 후에 저장하는 방식은 어떠신가요?
```suggestion
    public void approve(int registrationId) {
        final Registration registration = registration(registrationId);
        registration.approve();
        registrationRepository.save(registration);
    }
```

Reply:
수정했습니다.

상태를 변경해야하기 때문에 내부적으로 setter가 사용되었습니다.